### ISSUE TITLE
M4A4 AmmoInMagazine sometimes 0 after being bought

### ISSUE THREAD

**Issue by markus-wa** [markus-wa/demoinfocs-golang#149]
**Describe the bug**
Found by @BestAwperEver

See https://github.com/markus-wa/demoinfocs-golang/pull/147#issuecomment-530787344

> It is also a bit weird that sometimes M4A4 has a full ammo when bought and sometimes nothing at all: see lines 359, 362, 366, 426, 437 and so on.

~~This only seems to happen for `EqM4A4`~~ @BestAwperEver found another instance where it happens with the P250.

**To Reproduce**

See https://github.com/markus-wa/demoinfocs-golang/pull/147#issuecomment-530787344

**Expected behavior**
Ammo values are never 0 after being bought

**Library version**
v1.3.1

**Additional context**
-

**Comment by BestAwperEver** [comment]
Seems to be happening with p250 as well: see this match https://www.hltv.org/matches/2312372/big-vs-cloud9-pgl-major-krakow-2017
```
DEBUG: tick 116886: autimatic bought p250 AM:0 AR:0 AM2:0 AR2:0
DEBUG: tick 118894: Skadoodle bought p250 AM:0 AR:0 AM2:0 AR2:0
```
Might me some specific HLTV demo bug though.

**Comment by markus-wa** [comment]
Hmm, interesting. I wonder if there's a pattern with it only happening with certain weapons.

Note: The p250 and M4 both have alternative weapons, but maybe that's just coincidence.

**Comment by BestAwperEver** [comment]
I watched the demo and… that's it! 😄 The ones who had 0 ammo actually have bought CZ
```
DEBUG: tick 116886: autimatic bought p250 AM:0 AR:0 AM2:0 AR2:0
DEBUG: tick 117613: n0thing bought p250 AM:13 AR:26 AM2:13 AR2:26
DEBUG: tick 118894: Skadoodle bought p250 AM:0 AR:0 AM2:0 AR2:0
```
![image](https://user-images.githubusercontent.com/24738447/64861760-766f5b00-d639-11e9-8eb8-05e7b328370a.png)

**Comment by markus-wa** [comment]
Oh wow, thanks for the confirmation!

This should be an interesting one ...

**Comment by BestAwperEver** [comment]
Btw [this](https://github.com/markus-wa/demoinfocs-golang/issues/148#issuecomment-530944671) seems to be working:
```
DEBUG: tick 112596: keev bought p250 (P250) AM:13 AR:26 AM2:13 AR2:26
DEBUG: tick 116886: autimatic bought p250 (CZ75 Auto) AM:0 AR:0 AM2:0 AR2:0
DEBUG: tick 117613: n0thing bought p250 (P250) AM:13 AR:26 AM2:13 AR2:26
DEBUG: tick 118894: Skadoodle bought p250 (CZ75 Auto) AM:0 AR:0 AM2:0 AR2:0
```

**Comment by markus-wa** [comment]
Right, I saw that workaround, thanks 🙂. There is some similar code here already: https://github.com/markus-wa/demoinfocs-golang/blob/master/datatables.go#L457

I'll need to check if we can re-use that for #148 or if we should just take yours.

**Comment by BestAwperEver** [comment]
As a can see you choose an alternate weapon only if `strings.Contains(eq.OriginalString, altName)` returns true. The problem is -- at least with `PlayerHurt` event -- that original string for alternate weapons is (at least sometimes) empty so in the workaround I check for the default string. While it works it still doesn't seem right that only default weapons have proper original string.

**Comment by markus-wa** [comment]
Yeah I see why it happens, just need to find time to fix it 😅

`wepFix()` etc. just never gets used in `PlayerHurt`

**Comment by markus-wa** [comment]
If you'd like you could also create a PR, might be quicker!

**Comment by BestAwperEver** [comment]
I'd like to create a PR but I need some time to look deepr into the library code to understand exactly how it works (e. g. I don't understand for now from where `AmmoIn*` (not `*2` versions) comes from), hopefully I will have some free hours in the weekend 🙂

**Comment by markus-wa** [comment]
Sounds good.

`AmmoInMagazine` (as most other properties) comes from here: https://github.com/markus-wa/demoinfocs-golang/blob/master/datatables.go#L428
<sup>The `OnUpdate() / Bind()` design is questionable at best, but it is what it is (for now) 😄.</sup>

Feel free to ping me on Gitter if you need any help!

**Comment by markus-wa** [comment]
This will also be fixed by #150 as it's the same underlying issue.

**Comment by markus-wa** [comment]
Should be fixed with [v1.3.2](https://github.com/markus-wa/demoinfocs-golang/releases/tag/v1.3.2) 🙂
